libr/anal/fcn.c

```diff
@@ -1133,7 +1133,7 @@ static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut8 *buf, ut6
 			// if first byte in op.ptr is 0xff, then set leaddr assuming its a jumptable
 			{
 				ut8 buf[4];
-				anal->iob.read_at (anal->iob.io, op.ptr, &buf, sizeof (buf));
+				anal->iob.read_at (anal->iob.io, op.ptr, (ut8 *)&buf, sizeof (buf));
```

Comment:
`buf` is defined as an array of `ut8` literally in the line above, what's the added value of explicitly casting `&buf` to `ut8*`?

Reply:
Nothing :-) it s here to silent compilation warning with pedantic modern compiler.

Reply:
just pass buf without the & and the cast shouoldnt be necessary i think this patch is also in my kill-maxbb PR